### PR TITLE
Added picture links and links completion

### PR DESCRIPTION
**Pull request by asmfreak**
Hello! Thanks for the nice repo!
Here is a new extension for the wiki!

### CODE REVIEW

**Comment by TeddyDD** [comment]
Hi! Thanks for your contribution. I need to reviews your changes carefully since they add quite a lot of complexity. I noticed some issues:

- wiki command does not show completion
- after image expansion with `@` tag you can't go to new line with enter, you have to exit inset mode and enter it again

- not sure if `@+` syntax is good idea. Maybe ! should be modifier and user would create pages in the old way? WDYT?

File: wiki.kak [hunk 1]
```diff
@@ -13,35 +14,51 @@ define-command -hidden -params 1 wiki_setup %{
...
             execute-keys -draft %{
-                2h<a-b><a-k>\A@\w+<ret>
+                2h<a-b><a-k>\A@\+\w+<ret>
                 :wiki_expand_tag<ret>
             }
-            execute-keys <esc>hi
+            execute-keys <esc>h;di
         } }
     }
+    hook buffer InsertChar \n -group wiki %{
```
Comment (TeddyDD) [inline wiki.kak:42]: is separate hook necessary?

File: wiki.kak [hunk 2]
```diff
@@ -13,35 +14,51 @@ define-command -hidden -params 1 wiki_setup %{
 
 define-command wiki -params 1  \
 -docstring %{ wiki [file.md]: Edit or create wiki page } \
--shell-candidates %{ cd $kak_opt_wiki_path; find . -type f -name '*.md' | sed -e 's/^\.\///' }  \
+-shell-candidates %{ cd $kak_opt_wiki_path; find . -type f -name '*.md' | sed -e 's/^\.\///y' }  \
```
Comment (TeddyDD) [inline wiki.kak:17]: what does `y` option do in sed? I coudn't find docs for this

File: wiki.kak [hunk 3]
```diff
@@ -13,35 +14,51 @@ define-command -hidden -params 1 wiki_setup %{
...
+        } }
+    } 
     hook buffer NormalKey <ret> -group wiki %{
         wiki_follow_link
         wiki_toggle_checkbox
     }
+    set-option -add buffer completers option=wiki_completions
+    hook buffer -group wiki InsertIdle .* %{ try %{
+        execute-keys -draft <a-h><a-k>\B(@(\+|!)).\z<ret>
+        echo 'completing ...'
```
Comment (TeddyDD) [inline wiki.kak:58]: remove this echo, no need to be verbose

**Review by TeddyDD — CHANGES_REQUESTED** [review]
review

File: wiki.kak (hunk 2 above)
Comment (TeddyDD) [reply wiki.kak:17]: it seems that change breaks file completion in wiki command for me

**Comment by asmfreak** [comment]
I'll reiterate on this. I'm sorry. I've posted it half asleep, so it is very messy. I've spotted a very disappointing typo in README.md too.

**Comment by TeddyDD** [comment]
no worries, I have typos all over the place :) I really like idea of image links and completion, and it works quite well.  Actually only thing I don't like is `+` sign required for creating pages, but it shoudn't be too hard to change.

**Comment by TeddyDD** [comment]
I'm integrating some features from this pr in https://github.com/TeddyDD/kakoune-wiki/tree/update-07 branch:

- [x] expand picture command
-  [x] `@!image` syntax
- [ ] completion

**Comment by TeddyDD** [comment]
I'm rewriting this plugin from scratch. Closing this PR since it won't apply to new version.

**Pull request closed without merging**